libkbfs/folder_branch_ops.go

```diff
@@ -1100,30 +1100,65 @@ func (fbo *folderBranchOps) CheckForNewMDAndInit(
 			}()
 		}
 
-		if md.data.Dir.Type == Dir {
-			// this MD is already initialized
-			fbo.headLock.Lock(lState)
-			defer fbo.headLock.Unlock(lState)
-
-			// Only update the head the first time; later it will be
-			// updated either directly via writes or through the
-			// background update processor.
-			if fbo.head == nil {
-				err := fbo.setInitialHeadTrustedLocked(ctx, lState, md)
-				if err != nil {
-					return err
-				}
+		fbo.headLock.Lock(lState)
+		defer fbo.headLock.Unlock(lState)
+		// Only update the head the first time; later it will be
+		// updated either directly via writes or through the
+		// background update processor.
+		if fbo.head == (ImmutableRootMetadata{}) {
+			err = fbo.setInitialHeadTrustedLocked(ctx, lState, md)
+			if err != nil {
+				return err
 			}
-			return nil
 		}
-		// Initialize if needed
-		created = true
-		return fbo.initMDLocked(ctx, lState, md)
+		return nil
 	})
+}
+
+func (fbo *folderBranchOps) InitNewMD(ctx context.Context, id TlfID, handle *TlfHandle) (err error) {
```

Comment:
A function comment would be nice since this is called by other types.  It should say that this sets the MD head too, in addition to initializing it.